addons/product/static/src/js/product_configurator/product_configurator_dialog/product_configurator_dialog.js

```diff
@@ -0,0 +1,404 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { Component, onWillStart, useState, useSubEnv } from "@odoo/owl";
+import { Dialog } from '@web/core/dialog/dialog';
+import { ProductList } from "../product_list/product_list";
+import { useService } from "@web/core/utils/hooks";
+
+export class ProductConfiguratorDialog extends Component {
+    static components = { Dialog, ProductList};
+    static template = 'product.configuratorDialog';
+    static props = {
+        productTemplateId: Number,
+        ptavIds: { type: Array, element: Number },
+        customAttributeValues: {
+            type: Array,
+            element: Object,
+            shape: {
+                ptavId: Number,
+                value: String,
+            }
+        },
+        quantity: { type: Number, default: 0 },
+        productUOMId: { type: Number, optional: true },
+        companyId: { type: Number, optional: true },
+        edit: { type: Boolean, optional: true },
+        save: Function,
+        discard: Function,
+        close: Function, // This is the close from the env of the Dialog Component
+        options: {
+            type: Object,
+            shape: {
+                no_open: { type: Boolean, optional: true },
+                showCustom: { type: Boolean, optional: true },
+                showQty: { type: Boolean, optional: true },
+                qtyFieldName: {
+                    type: String,
+                    optional: true,
+                }, // Some models have different fields names for qty
+                productUomIdFieldName: {
+                    type: String,
+                    optional: true,
+                }, // Some models have different fields names for product_uom
```

Comment:
I think Guru's talked about that because your solution is indeed "cleaner", but creating new modules just to redefine a method seemed overkill just for a field name change.

Reply:
I don't want you to create new modules. I'm aware of the decision to reduce the number of modules and I think it's a good decision. I just want to follow one rule: If anything added is used only once, it should not be in the product configurator but should be a customization in your module(s).

Let me rephrase my precedent comment:
Why do you have `qtyFieldName` and `productUomIdFieldName` in the options of `product_configurator_dialog` but never use them anywhere in those files?